### ISSUE TITLE
Wrong type in openapi-react-query when using select to transform data

### ISSUE THREAD

**Issue by HagenMorano** [openapi-ts/openapi-typescript#1845]
*Mind: This is a bug in openapi-react-query. There is no template for opening Bugs for the openapi-react-query package yet.*

**Description**

The `select` transformer in the `queryOptions` of the useQuery method expects a value of type `data` as return value. The value should be transformable to an arbitrary value (and be inherited correctly). Example:

Cool:
```ts
$api.useQuery('get', '/pets/person', {},
  {
    select: (data) => data,
  }
);
```
Not cool:
```ts
$api.useQuery('get', '/pets/person', {},
  {
    select: (data) => data.age, // or any other value
  }
);
```

**Reproduction**

See the implementation at the bottom of `fetchClient.ts` in this  [Stackblitz](https://stackblitz.com/edit/vitejs-vite-gwavds?file=src%2FfetchClient.ts).

**Expected result**

Arbitrary return values in the `select` method do not throw type errors. Furthermore, `data` should infer the type of the returned value of the `select` method (if set):

```ts
const { data } = $api.useQuery('get', '/pets/person', {},
  {
    select: (data) => data.age, // this shouldn't throw any type errors
  }
);

const str: number | undefined = data; // this shouldn't throw any type errors either as date.age is of type number and optional.
```

**Comment by drwpow** [comment]
Thanks for filing! You’re right—we should add a template for openapi-react-query. Though for now, it’s not an issue opening a bug against openapi-fetch because there’s still overlap in what those 2 libraries do.

**Comment by kerwanp** [comment]
Thanks for the detailed answer, this definitely needs to be addressed. I'll try to find some time in the following days to improve the types of the libraries and use more properly the @tanstack/reat-query (and hopefully fix this issue).

As you are already advanced on this subject, I am totally open to review a PR that fixes this issue.
